code/obj/machinery/phosphorizer.dm

```diff
@@ -0,0 +1,163 @@
+		phos_target.color = "#[colr2][colg2][colb2]"
+
+		var/nameadjust = istype_exact(phos_target,/obj/item/light/tube) ? "tube" : "bulb"
+
+		phos_target.name = "phosphorized light [nameadjust]"
+		phos_target.desc = "A light [nameadjust] that has been coated with a phosphor to change its hue. A small label is marked '[ctrl_R]-[ctrl_G]-[ctrl_B]'."
+
+		phos_target.pixel_x = -4
+		phos_target.pixel_y = -8 //move bulb more towards output chute
+		phos_target.set_loc(src.loc)
+
+		playsound(src.loc, sound_process, 80, 1)
+
+	proc/stop_phos()
+		src.phosphorizing = false
+		UpdateOverlays(null, "operatebar", 0, 1)
+
+	proc/start_phos()
+		src.phosphorizing = true
+		var/image/O_panel = SafeGetOverlayImage("operatebar", 'icons/obj/machines/phosphorizer.dmi', "activelight")
+		UpdateOverlays(O_panel, "operatebar", 0, 1)
+		sleep(phos_delay)
+
+		for (var/obj/item/M in src.contents)
+			if(status & NOPOWER || src.phosphorizing == false)
+				failbreak = true
+				break
+			use_power(src.power_usage)
+			colorize_bulb(M)
+
+			sleep(phos_delay)
+
+		if(failbreak)
+			failbreak = false
+			src.visible_message("<b>[src]</b> abruptly stops operating.")
+			playsound(src.loc, sound_grump, 20, 1)
+		else
+			src.visible_message("<b>[src]</b> finishes working and shuts down.")
+			playsound(src.loc, sound_happi, 20, 1)
+		if(src.phosphorizing) stop_phos()
+
+/obj/machinery/phosphorizer/power_change()
+	var/image/I_panel = SafeGetOverlayImage("statuspanel", 'icons/obj/machines/phosphorizer.dmi', "powerpanel")
+	I_panel.plane = PLANE_SELFILLUM
+	I_panel.alpha = 128
+	if (status & BROKEN)
+		UpdateOverlays(null, "statuspanel", 0, 1)
+		if(src.phosphorizing) src.stop_phos()
+	else
+		if ( powered() )
+			UpdateOverlays(I_panel, "statuspanel", 0, 1)
+			status &= ~NOPOWER
+		else
+			SPAWN_DBG(rand(0, 5))
+				UpdateOverlays(null, "statuspanel", 0, 1)
+				status |= NOPOWER
+
+/obj/machinery/phosphorizer/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "Phosphorizer", name)
+		ui.open()
+
+/obj/machinery/phosphorizer/ui_data(mob/user)
+	. = list(
+		"lights" = src.contents.len,
+		"online" = src.phosphorizing,
+		"hostR" = src.ctrl_R,
+		"hostG" = src.ctrl_G,
+		"hostB" = src.ctrl_B,
+	)
+
+/obj/machinery/phosphorizer/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if (.)
+		return
+	switch(action)
+		if("tune_hue")
+			switch(params["hue"])
+				if("R") src.ctrl_R = clamp(round(params["output"]), 20, 255)
+				if("G") src.ctrl_G = clamp(round(params["output"]), 20, 255)
+				if("B") src.ctrl_B = clamp(round(params["output"]), 20, 255)
+			. = TRUE
+		if("toggle-process")
+			if(status | BROKEN && powered() && src.contents.len)
+				if(!src.phosphorizing)
+					src.start_phos()
+					boutput(usr, "You activate [src].")
+				else
+					src.stop_phos()
+					boutput(usr, "You deactivate [src].")
+				. = TRUE
+		if("eject")
+			if(src.contents.len)
```

Comment:
```suggestion
			if(length(src.contents))
```
Prefer `length(foo)` over `foo.len`.

Reply:
Yep, sure. No instances of .len in the file now. f7358dc, as with the rest of this batch.